src/main/java/duke/storage/Storage.java

```diff
@@ -0,0 +1,116 @@
+package duke.storage;
+
+import duke.exception.DukeException;
+import duke.tasklist.TaskList;
+import duke.tasks.Deadline;
+import duke.tasks.Event;
+import duke.tasks.Task;
+import duke.tasks.ToDo;
+
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
+import java.nio.file.Files;
+import java.nio.file.Paths;
+
+import java.time.LocalDate;
+import java.time.LocalTime;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/** This class deals with loading tasks from the file and saving tasks in the file */
+public class Storage {
+    private String path;
+    private BufferedWriter writer;
+
+    /**
+     * Constructs a Storage object with a path as the given path and a BufferedWriter object
+     * that writes to the file located at the given path.
+     * @param path The path of the storage file.
+     */
+    public Storage(String path) throws IOException {
+        this.path = path;
+        File file = new File(path);
+        if (!Files.exists(Paths.get("./data"))) {
+            new File("./data").mkdir();
+        }
+        file.createNewFile();
+        this.writer = new BufferedWriter(new FileWriter(path, true));
+    }
+
+    /**
+     * Loads the tasks from the file.
+     * @return A List containing the saved tasks.
+     * @throws DukeException An exception while loading tasks from the file.
+     */
+    public List<Task> load() throws IOException, DukeException {
```

Comment:
I'm not sure if this might help, but would it be good to include line breaks between logical pieces of code in this fxn to improve readability? :D